tools/wff-optimizer/src/test/java/com/google/android/clockwork/wff/optimizer/ImageLoaderTest.kt

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.clockwork.wff.optimizer
+
+import com.google.common.truth.Truth.assertThat
+import java.awt.image.BufferedImage
+import java.io.File
+import java.io.StringWriter
+import java.util.HashSet
+import javax.xml.parsers.DocumentBuilderFactory
+import javax.xml.transform.TransformerFactory
+import javax.xml.transform.dom.DOMSource
+import javax.xml.transform.stream.StreamResult
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.w3c.dom.Document
+
+@RunWith(JUnit4::class)
+class ImageLoaderTest {
+    @Test
+    fun computeNonTransparentBounds() {
+        val imageLoader =
+            ImageLoader(Settings(File("src/test/resources/imageTests").getAbsolutePath()))
+
+        val image = imageLoader.loadImage("alpha")
+
+        assertEquals(image.nonTransparentBounds!!.left, 244)
+        assertEquals(image.nonTransparentBounds!!.right, 380)
+        assertEquals(image.nonTransparentBounds!!.top, 330)
+        assertEquals(image.nonTransparentBounds!!.bottom, 415)
+    }
+
+    @Test
+    fun bitmapFontImageDeduplication() {
+        val builder = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+        val settings = Settings(File("src/test/resources/duplicateImageTest").getAbsolutePath())
+        val document =
+            builder.parse(File("src/test/resources/duplicateImageTest/res/raw/watchface.xml"))
+        val imageLoader = TestImageLoader(settings)
+        val optimizer = Optimizer(document, settings)
+        optimizer.walkTree()
+        optimizer.bitmapFonts.optimize(imageLoader)
+
+        assertTrue(imageLoader.dedupeAndWriteOptimizedImages())
+
+        // This should deduplicate file_a1, file_a2,file_a3 which are all the same, likewise file_b1
+        // and file_b2 are the same.
+        assertEquals(
+            """<?xml version="1.0" encoding="UTF-8" standalone="no"?><!-- Copyright (C) 2024 Google LLC
```

Comment:
This file is very busy. What about moving the expected xml into a file next to the source one? At that point `diff` the actual and the expected file becomes easy. 

* watchface.xml 
* watchface.xml_expected

I suspect that you could then replace most of the test code with a function call as they look pretty similar to each other.

Reply:
Done